spec/requests/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe API::V1::UsersController, type: :request do
+  context 'when a user registers' do
+    it 'returns the user' do
+      post :create, params: create_user_params.merge!(email: 'new_email@gmail.com')
```

Comment:
We dont need `!`, please check bellow test cases also 👍 
```suggestion
      post :create, params: create_user_params.merge(email: 'new_email@gmail.com')
```

Reply:
@ankitkalia1195 interesting feedback from Junan, what do you think? :) 

I do not have any problem by updating to the no-bang method (as discussed with Junan, performances won't change much in a single-call and so-small hash like this), but I'm not convinced 100% by the risk of this `merge!` in this specific context. 
It's also to better understand the general idea of when does it makes sense to risk the bang-method and when we should prefer the non-bang ones :) 
Thanks

Reply:
But I guess for this comment from junan makes sense, by default it should be merge(less risk) and when you require to make change to the same hash, then it should be merge!. Altough in your case there is no risk, but making a habit to use ! can be risky in this situation:

```
  params = user_params

  post :some_action, params: params.merge!(a: 'b')

  post :another_action, params: params.merge!(b: 'c')

```

Or image you are passing hash as argument to some function, where you unintentionally modify hash without :

```
  def do_something(params)
    params.merge!(a: 'b')
  end

  do_something(params)

```

So to summarize:

```
  * Use bang method when you actually need to modify the same hash object in place
  * Use non-bang method otherwise

```

Reply:
Ok, it's similar to Junan feedbacks in another discussion~ So I need to change my habits :P 
Thanks guys! And sorry Junan for having been a little bit sceptical, it change a lot from what I used to do~

Reply:
Ok solved ;-)

Reply:
@malparty Its fine, actually it is good practice to always verify the changes rather than just following feedback 👍